renku/ui/service/views/error_handlers.py

```diff
@@ -375,8 +378,26 @@ def decorated_function(*args, **kwargs):
     return decorated_function
 
 
-def handle_migration_errors(f):
-    """Wrapper which handles migrations exceptions."""
+def handle_migration_read_errors(f):
+    """Wrapper which handles migrations read exceptions."""
+    # noqa
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        """Represents decorated function."""
+        # NOTE: verify if this may better go in MigrationsCheckCtrl as try/except in to_response()
+        try:
+            return f(*args, **kwargs)
+        except TemplateMissingReferenceError as e:
+            raise UserProjectTemplateReferenceError(e)
+        except (InvalidTemplateError, TemplateUpdateError) as e:
+            raise IntermittentProjectTemplateUnavailable(e)
+
+    return decorated_function
+
+
+@handle_migration_read_errors
+def handle_migration_write_errors(f):
```

Comment:
Why do we need to distinguish between migration read/write errors. AFAIK, the difference is that in read we only check for migration; so, the same errors can happen in both cases.

Reply:
The write section handles a couple of extra errors, and those throw a more severe error (`Program` instead of `Intermittent`).
The point is that errors while reading are somewhat expected (e.g. template may be deleted, that's not a bug), while actually migrating the project should be invoked by the client only if the read errors didn't happen.
So a `DockerfileUpdateError` or a `MigrationError` caught on writing is likely to be a bug or severe error from which the user may not be able to recover in the UI.